DevelopmentProposals/reporterColumnLabels/reporterColumnLabels.md

```diff
@@ -54,14 +54,15 @@ and forcing the user to specify an annotation for each Output may not be practic
 - The TableReporter should use a default column name that is very likely to be meaningful.
 
 ### Architecture
-The selected design involves four main changes:
+The selected design involves six main changes:
 
 1. Rename "annotation" to "alias" (e.g., `AbstractInput::getAnnotation()` becomes `AbstractInput::getAlias()`).
 2. If the user provides the optional second argument when connecting an Output to an Input, store the string as the Input's alias; otherwise, the alias remains null.
-3. Create `getShortLabel()` and `getLongLabel()` methods on Input:
-  - `getShortLabel()` will be called by ConsoleReporter and will return `isNull(alias) ? output_name : alias`.
-  - `getLongLabel()` will be called by TableReporter and will return `isNull(alias) ? full_path_name : alias`.
-4. Create `setAlias()` methods analogous to the current `getAnnotation()` methods so the user can change the alias after (but not before) connecting.
+3. Create `getReportedLabel()` on Input, which will always return the alias if one is provided:
+  - `getReportedLabel()` will be called by a Reporter and will return `isNull(alias) ? full_path_name : alias`.
```

Comment:
> Why would a user not request the *alias* directly that they assigned and use either `getShortName()` or `getLongName()` which will give the same value if an alias is provided? What are the non Reporter use cases for `getShortLabel()` and `getLongLabel()`?

Potential use cases include displaying information about connections in the GUI and generating plot legends. It was my understanding that we wanted to avoid code like `text = !getAlias().empty() ? getAlias() : getOutput().getName()` and path string parsers outside Input. Instead, Input would be asked for a short or long label and the logic for determining the short and long versions of the label would be centralized in Input.

> My interpretation of the meeting was that short and long name were used to refer to the alias or output name and alias or full_path_name in the discussion and not that they would be the actual method names. When I viewed the revised DevProp I thought the names were stand ins to be revised.

I thought the devprop was very clear about what the methods would be named (see item 3: "Create `getShortLabel()` and `getLongLabel()` methods on Input").

> I was surprised when the DevProp got merged today without review and that you had a PR in.

I was unaware that there were any outstanding design concerns following the Dev Team meeting. @chrisdembia reviewed and merged and devprop, which reflected what seemed to be a unanimous agreement on the design. Apparently I misinterpreted the conversation during the meeting, or perhaps there was a follow-up discussion that I did not hear and that was not summarized in a comment on the devprop PR.

> If you believe that I raise valid points then I will make the changes

I don't think that is how this process is supposed to work. I will fix the implementation once we all agree on a new design.